python_modules/dagster/dagster/_config/pythonic_config/resource.py

```diff
@@ -867,16 +870,18 @@ def separate_resource_params(cls: Type[BaseModel], data: Dict[str, Any]) -> Sepa
     are marked as resources (ie, using ResourceDependency) from those which are not.
     """
     keys_by_alias = {field.alias: field for field in cls.__fields__.values()}
-    data_with_annotation: List[Tuple[str, Any, Type]] = [
+    data_with_annotation: List[Tuple[str, Any, Type, List[str]]] = [
         # No longer exists in Pydantic 2.x, will need to be updated when we upgrade
-        (k, v, keys_by_alias[k].outer_type_)
+        (k, v, keys_by_alias[k].outer_type_, [])
         for k, v in data.items()
         if k in keys_by_alias
     ]
     out = SeparatedResourceParams(
-        resources={k: v for k, v, t in data_with_annotation if _is_annotated_as_resource_type(t)},
+        resources={
+            k: v for k, v, t, m in data_with_annotation if _is_annotated_as_resource_type(t, m)
+        },
         non_resources={
-            k: v for k, v, t in data_with_annotation if not _is_annotated_as_resource_type(t)
+            k: v for k, v, t, m in data_with_annotation if not _is_annotated_as_resource_type(t, m)
         },
     )
     return out
```

Comment:
Not needed in this PR, but used downstream. Can move it there if that's better. We need the metadata to tell if this was a `typing.Annotated` annotation (which we use for resource/resource deps), since Pydantic 2.0 strips that info out and sticks its metadata in the metadata on the field here.

Reply:
Ok just copy and paste this commentary into a comment